lib/std/secrets.nim

```diff
@@ -0,0 +1,220 @@
+                            BCRYPT_USE_SYSTEM_PREFERRED_RNG)
+
+  proc urandom*(p: var openArray[byte]): int =
+    let size = p.len
+    if size > 0:
+      result = randomBytes(addr p[0], size)
+
+elif defined(linux):
+  let SYS_getrandom {.importc: "SYS_getrandom", header: "<sys/syscall.h>".}: clong
+
+  proc syscall(n: clong, buf: pointer, bufLen: cint, flags: cuint): int {.importc: "syscall", header: "<sys/syscall.h>".}
+
+  proc randomBytes(p: pointer, size: Natural): int =
+    while result < size:
+      let readBytes = syscall(SYS_getrandom, p, cint(size - result), 0)
+      processReadBytes(readBytes, p, result)
+
+  proc urandom*(p: var openArray[byte]): int =
+    let size = p.len
+    if size > 0:
+      result = randomBytes(addr p[0], size)
+
+elif defined(openbsd):
+  proc getentropy(p: pointer, size: cint): cint {.importc: "getentropy", header: "<unistd.h>".}
+
+  proc randomBytes(p: pointer, size: Natural): int =
+    while result < size:
+      let readBytes = getentropy(p, cint(size - result))
+      processReadBytes(readBytes, p, result)
+
+  proc urandom*(p: var openArray[byte]): int =
+    let size = p.len
+    if size > 0:
+      result = randomBytes(addr p[0], size)
+
+elif defined(freebsd):
+  type ssize_t = int
+  proc getrandom(p: pointer, size: csize_t, flags: cuint): ssize_t {.importc: "getrandom", header: "<sys/random.h>".}
+
+  proc randomBytes(p: pointer, size: int): int =
+    while result < size:
+      let readBytes = getrandom(p, csize_t(size - result), 0)
+      processReadBytes(readBytes, p, result)
+
+  proc urandom*(p: var openArray[byte]): int =
+    let size = p.len
+    if size > 0:
+      result = randomBytes(addr p[0], size)
+
+elif defined(ios):
+  {.passL: "-framework Security".}
+
+  const errSecSuccess = 0
+  type
+    SecRandom {.importc: "struct __SecRandom".} = object
+
+    SecRandomRef = ptr SecRandom
+      ## An abstract Core Foundation-type object containing information about a random number generator.
+
+  proc secRandomCopyBytes(
+    rnd: SecRandomRef, count: csize_t, bytes: pointer
+    ): cint {.importc: "SecRandomCopyBytes", header: "<Security/SecRandom.h>".}
+
+  proc urandom*(p: var openArray[byte]): int =
+    let size = p.len
+    if size > 0:
+      # `kSecRandomDefault` is a synonym for NULL.
+      result = secRandomCopyBytes(nil, csize_t(size), addr p[0])
+
+elif defined(macosx):
+  proc getentropy(p: pointer, size: csize_t): cint {.importc: "getentropy", header: "<sys/random.h>".}
+
+  proc randomBytes(p: pointer, size: Natural): int =
+    while result < size:
+      let readBytes = getentropy(p, csize_t(size - result))
+      processReadBytes(readBytes, p, result)
+
+  proc urandom*(p: var openArray[byte]): int =
+    let size = p.len
+    if size > 0:
+      result = randomBytes(addr p[0], size)
+else:
+  proc urandom*(p: var openArray[byte]): int =
+    let size = p.len
+    if size > 0:
+      result = getDevUrandom(p, size)
+
```

Comment:
change all `proc urandom*(p: var openArray[byte]): int =` procs to `urandomImpl` and then introduce:
```nim
proc urandom*(p: var openArray[byte]): int {.inline.} =
  ## doc comment... in particular explaning return value
  result = urandomImpl(p)
```
so that it can have a doc comment (`when defined nimdoc` is possible too but this is simpler IMO)